### PR TITLE
Test related to authentication with x.509 certificates.

### PR DESCRIPTION
**Pull request by lukasz-soszynski-eliatra**
### Description
[Describe what this change achieves]
* Category (Integration tests)
* Why these changes are required?
* What is the old behavior before changes and new behavior after changes?

### Issues Resolved
[List any issues this PR will resolve]

Integration tests related to user authentication with X.509 certificates.

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — DISMISSED** [review]
LGTM!

**Comment by DarshitChanpura** [comment]
Bwc tests are broken on main. Tracking issue here: https://github.com/opensearch-project/security/issues/2221

File: src/integrationTest/java/org/opensearch/security/http/CertificateAuthenticationTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		try (TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+
+			HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(SC_OK);
+		}
+	}
+
+	@Test
+	public void shouldAuthenticateUserWithCertificate_positiveUserSpoke() {
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/http/CertificateAuthenticationTest.java:85]: nit: Spock

File: src/integrationTest/java/org/opensearch/security/http/CertificateAuthenticationTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+	private static final String BACKEND_ROLE_BRIDGE = "bridge";
+	private static final String BACKEND_ROLE_CAPTAIN = "captain";
+
+	private static final Role ROLE_ALL_INDEX_SEARCH = new Role("all-index-search").indexPermissions("indices:data/read/search")
+		.on("*");
+
+	private static final Map<String, Object> CERT_AUTH_CONFIG = Map.of(
+		"username_attribute", "cn",
+		"roles_attribute", "ou"
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/http/CertificateAuthenticationTest.java:59]: Interesting, I didn't know `roles_attribute` was possible with client cert authentication. I don't see documentation for this on the documentation website: https://opensearch.org/docs/latest/security-plugin/configuration/client-auth/

@cwillum, putting this on your radar to visit when client cert authentication is re-visited on the documentation website.

My understanding of Organizational Unit in LDAP is that it is conventionally used to model hierarchical structures of organizations. i.e. As a contributor the OpenSearch project concentrated on security, I may model a Distinguished name as:

```
L=NYC,CN=Craig,OU=OpenSearch,OU=Security,DC=opensearch,DC=org
```

Here it looks like we are using this as the list of backend roles and using that it is repeated as a convenience.

**Review by cwperks — APPROVED** [review]
Thank you @lukasz-soszynski-eliatra, this looks good to me. There may be an opportunity to test a scenario with multiple backend roles, but I think this adds some nice coverage!

**Review by DarshitChanpura — APPROVED** [review]

**Pull request merged**